rest-fetcher/src/main/java/com/ackee/versionupdatehandler/RestVersionFetcher.kt

```diff
@@ -0,0 +1,70 @@
+package com.ackee.versionupdatehandler
+
+import com.ackee.versioupdatehandler.VersionFetcher
+import com.ackee.versioupdatehandler.model.BasicVersionsConfiguration
+import com.ackee.versioupdatehandler.model.DefaultVersionsConfiguration
+import com.ackee.versioupdatehandler.model.VersionsConfiguration
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonObject
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+
+/**
+ * Class that fetches version configuration from rest api
+ *
+ *
+ * Api is specified by base url and is expected to be at address /app_version
+ *
+ * @author David Bilik [david.bilik@ackee.cz]
+ * @since 06/02/2017
+ */
+class RestVersionFetcher constructor(
+    private val baseUrl: String,
+    private val currentAttributeName: String = CURRENT_VERSION,
+    private val minimalAttributeName: String = MINIMAL_VERSION
+) : VersionFetcher {
+
+    internal interface ApiDescription {
+        @GET("app_version")
+        suspend fun versions(): JsonObject?
+    }
+
+    private var api: ApiDescription? = null
+        get() {
+            if (field == null) {
+                field = Retrofit.Builder()
+                    .baseUrl(baseUrl)
+                    .addConverterFactory(
+                        GsonConverterFactory.create(GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create()))
+                    .client(OkHttpClient.Builder()
+                        .addNetworkInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
```

Comment:
remove this pls. We don't want our library to add any logs that cannot be controlled from the outside

Reply:
d47f415d7dd7ad63d34c8478dd5f273cbf58d87d